src/cfn-mcp-server/awslabs/cfn_mcp_server/stack_analysis/cloud_formation_utils.py

```diff
@@ -0,0 +1,178 @@
+        """
+        response = self.cfn_client.list_stack_resources(StackName=stack_name)
+        return response.get('StackResourceSummaries', [])
+
+    def get_stack_template(self, stack_name: str) -> Dict[str, Any]:
+        """Get the template for a CloudFormation stack.
+
+        Args:
+            stack_name: Name of the stack
+
+        Returns:
+            Dict containing the stack template
+        """
+        response = self.cfn_client.get_template(StackName=stack_name)
+        return response.get('TemplateBody', {})
+
+    # Resource Scan API methods
+    def start_resource_scan(self) -> str:
+        """Start a new resource scan and return the scan ID.
+
+        Returns:
+            Resource scan ID
+        """
+        try:
+            logger.info('Starting resource scan...')
+            response = self.cfn_client.start_resource_scan()
+            scan_id: str = response['ResourceScanId']
+            self.resource_scan_id = scan_id
+            logger.info(f'Resource scan started with ID: {scan_id}')
+            return scan_id
+        except Exception as e:
+            logger.error(f'Error starting resource scan: {str(e)}')
+            raise handle_aws_api_error(e)
+
+    def get_resource_scan_status(self, scan_id: Optional[str] = None) -> Dict[str, Any]:
+        """Get the status of a resource scan.
+
+        Args:
+            scan_id: Resource scan ID (uses stored ID if not provided)
+
+        Returns:
+            Resource scan status information
+        """
+        if scan_id is not None:
+            effective_scan_id = scan_id
+        elif self.resource_scan_id is not None:
+            effective_scan_id = self.resource_scan_id
+        else:
+            raise ClientError('No resource scan ID available')
+
+        try:
+            response = self.cfn_client.describe_resource_scan(ResourceScanId=effective_scan_id)
+            return response
+        except Exception as e:
+            logger.error(f'Error getting resource scan status: {str(e)}')
+            raise handle_aws_api_error(e)
+
+    def list_resource_scan_resources(self, scan_id: Optional[str] = None) -> List[Dict[str, Any]]:
+        """List all resources found in the resource scan.
+
+        Args:
+            scan_id: Resource scan ID (uses stored ID if not provided)
+
+        Returns:
+            List of all resources found in the scan
+        """
+        if scan_id is not None:
+            effective_scan_id = scan_id
+        elif self.resource_scan_id is not None:
+            effective_scan_id = self.resource_scan_id
+        else:
+            raise ClientError('No resource scan ID available')
+
+        try:
+            logger.info(f'Listing resources from scan {effective_scan_id}')
+
+            # Use paginator to handle large result sets
+            paginator = self.cfn_client.get_paginator('list_resource_scan_resources')
+            page_iterator = paginator.paginate(ResourceScanId=effective_scan_id)
```

Comment:
wonder if this method should return the iterator so Q (and MCP's memory) doesn't get overloaded from a big scan and can work in batches/chunks